### PR TITLE
Refactor part 1: reorganize dir, split up utilities, split up components

### PR DESCRIPTION
**Pull request by iliketigers**

### CODE REVIEW

File: server/index.js [hunk 1]
```diff
@@ -17,21 +17,12 @@ app.get('/prices/:ticker/:date', async (request, response) => {
     const apiUrl = `https://api.tiingo.com/tiingo/daily/${ticker}/prices?startDate=${apiDate}&endDate=${apiDate}&token=${tiingo_api_key}`;
     const fetchResponse = await fetch(apiUrl);
```
Comment (kamahen) [inline server/index.js:18]: I've used this; don't know if it's better or worst (and might even be wrong):
```
fetch(apiURL)
  .then(response => response.json())
  .then(callback)  // not sure what your callback is here
  .catch(err => { console.log(err); };  // TODO: maybe this goes before the callback?
```
I haven't found a definitive answer for how best to do this kind of thing. :(

File: src/components/EditableStockListRow.js [hunk 2]
```diff
@@ -0,0 +1,106 @@
+import React, { Component } from 'react';
+import * as Utilities from '../utils/utilities';
+import * as ApiCalls from '../utils/apiCalls';
+
+export class EditableStockListRow extends Component {
+  state = {};
+  handleClick = (name, id, symbol, shares) => {
+    if (name === 'delete') {
+      this.props.onDelete(id);
+    }
```
Comment (kamahen) [inline src/components/EditableStockListRow.js:10]: Suggestion:
```
if (name == 'delete') {
  ...
} else if (name == 'save') {
  ...
} else if ...
  ...
}
```
is a bit clearer what's going on and also let's you put in a final `else` to catch anything else (which would be an error , probably).

Another way of doing this is to create an map of functions (untested):
```
const handlers = {
   'delete': (id, symbol, shares) => { this.props.onDelete(id); },
   'save': (id, symbol, shares) => { this.props.onSaveOrCancel(id, symbol, shares); }
   'cancel': (id, symbol, shares) => this.props.onSaveOrCancel(); }
};
handleClick = (name, id, symbol,s hares) => {
    handlers[name](id, symbol, shares);
}
```

File: src/utils/chartUtils.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
+import * as DateUtils from './dateUtils';
+import * as ApiCalls from './apiCalls';
+
+/**
+ *
+ * @param {Object} startDate
```
Comment (kamahen) [inline src/utils/chartUtils.js:6]: Should `Object` be `Date`?

File: src/utils/apiCalls.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+      xAxisLabels: [],
+      yAxisLabels: [],
+      name: `Invalid ticker ${ticker}`,
+    };
+  }
+  if (!pricesJson) {
+    return;
+  }
+
+  for (let i = 0; i < pricesJson.length; i++) {
```
Comment (kamahen) [inline src/utils/apiCalls.js:53]: The `entries()` method gives an iterator of [index, item] which can be pattern-matched, so:
```
for (const [i, price] of pricesJson.entries()) {
    const copyOfDate = DateUtils.convertPickedDateToUtc(date);
    const apiDate = price.date;
    if (apiDate.toISOString() === copyOfDate.toISOString()) {
      return price.close;
    }
}
```
except in this case you don't need `i`, so can just do `for (price of pricesJson)`.

And if you only want the indexes, you can do `for (i in pricesJson)`.

File: server/index.js (hunk 1 above)
Comment (iliketigers) [reply server/index.js:18]: async await is intended to replace .then in most cases

File: src/utils/chartUtils.js (hunk 3 above)
Comment (iliketigers) [reply src/utils/chartUtils.js:6]: hmm... i thought jsdoc only accepted basic types (string/object/bool/etc) but I can't find anything that says it's bad so sure

File: server/index.js (hunk 1 above)
Comment (iliketigers) [reply server/index.js:18]: https://javascript.info/async-await

File: src/components/EditableStockListRow.js (hunk 2 above)
Comment (iliketigers) [reply src/components/EditableStockListRow.js:10]: oh i like that, fixing in next push

Comment (iliketigers) [reply src/components/EditableStockListRow.js:10]: filed #31

File: src/utils/apiCalls.js (hunk 4 above)
Comment (iliketigers) [reply src/utils/apiCalls.js:53]: filed #31

**Pull request merged**